generator/codegen/linkfollower_extensions.go

```diff
@@ -0,0 +1,392 @@
+		return
+	}
+
+	// Return just one type with no paranthesis or name
+	var buf bytes.Buffer
+	if len(fields.List) == 1 && len(fields.List[0].Names) == 0 {
+		printer.Fprint(&buf, fset, fields.List[0].Type)
+		src = buf.String()
+		return
+	}
+
+	// Process multiple return types, named and unnamed
+	buf.WriteByte('(')
+	for i, field := range fields.List {
+		for j, name := range field.Names {
+			buf.WriteString(name.Name)
+			if j < len(field.Names)-1 {
+				buf.WriteByte(',')
+			} else {
+				buf.WriteByte(' ')
+			}
+		}
+
+		printer.Fprint(&buf, fset, field.Type)
+
+		if i < len(fields.List)-1 {
+			buf.WriteString(", ")
+		}
+	}
+	buf.WriteByte(')')
+
+	src = buf.String()
+	return
+}
+
+func modifyFirstParam(params *ast.FieldList) (err error) {
+	if params == nil || len(params.List) == 0 {
+		err = commonerrors.Newf(commonerrors.ErrUnexpected, "Could not modify first parameter, no fields exists")
+		return
+	}
+
+	param := params.List[0]
+
+	if _, ok := param.Type.(*ast.StarExpr); !ok {
+		param.Type = &ast.StarExpr{
+			Star: token.NoPos,
+			X:    param.Type,
+		}
+	}
+
+	return
+}
+
+func getFirstParam(params *ast.FieldList, fset *token.FileSet) (paramName string, paramType string, err error) {
+	if params == nil || len(params.List) == 0 {
+		err = commonerrors.Newf(commonerrors.ErrUnexpected, "Could not render source code for first parameter, no fields exists")
+		return
+	}
+
+	param := params.List[0]
+
+	if len(param.Names) > 1 {
+		err = commonerrors.Newf(commonerrors.ErrUnexpected, "Could not get first parameter, expected only one name")
+		return
+	}
+
+	if len(param.Names) == 0 {
+		err = commonerrors.Newf(commonerrors.ErrUnexpected, "Could not get first parameter, expected a named parameter")
+		return
+	}
+
+	paramName = param.Names[0].Name
+	var buf bytes.Buffer
+	printer.Fprint(&buf, fset, param.Type)
+	paramType = buf.String()
+
+	return
+}
+
+func getReceiverType(fn *ast.FuncDecl) (reciever string, err error) {
+	if fn.Recv == nil || len(fn.Recv.List) == 0 {
+		err = commonerrors.Newf(commonerrors.ErrUnexpected, "No receiver defined for func `%s()`", fn.Name.Name)
+		return
+	}
+
+	expr := fn.Recv.List[0].Type
+
+	switch t := expr.(type) {
+	case *ast.Ident:
+		reciever = t.Name
+		return
+	case *ast.StarExpr:
+		if ident, ok := t.X.(*ast.Ident); ok {
+			reciever = ident.Name
+			return
+		}
+	}
+
+	err = commonerrors.Newf(commonerrors.ErrUnexpected, "Could not get receiver for func `%s()`", fn.Name.Name)
+	return
+}
+
+func modifyLocalVarPath(fn *ast.FuncDecl) (err error) {
+	for _, stmt := range fn.Body.List {
+		assign, ok := stmt.(*ast.AssignStmt)
+		if !ok {
+			continue
+		}
+
+		if assign.Tok == token.DEFINE && len(assign.Lhs) > 0 {
+			if ident, ok := assign.Lhs[0].(*ast.Ident); ok && ident.Name == "localVarPath" {
+				if expr, ok := assign.Rhs[0].(*ast.BinaryExpr); ok && len(assign.Rhs) > 0 {
+					expr.Y = ast.NewIdent("link")
```

Comment:
could the value "link" be defined as a constant  so that if we wanted to change the content to e.g. url/endpoint/... we would only need to change it in one place